File: solutions/1019-squares-of-a-sorted-array/solution.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int ans[] = new int[nums.length];
//         make square
        for(int i=0;i<nums.length;i++){
            nums[i] = nums[i]*nums[i];
        }
        
        int i=0;
        int j=nums.length-1;
        int c=nums.length-1;
        
        while(i<j){
            if(nums[i]>=nums[j]){
                ans[c] = nums[i];
                i++;
                c--;
            }
            else{
                ans[c] = nums[j];
                j--;
                c--;
            }
        }
        ans[0] = nums[i];
        return ans;
    }
}

File: solutions/1019-squares-of-a-sorted-array/solution.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int i=0;
        int j=nums.size()-1;
        int index = nums.size()-1;
        vector<int> pums(nums.size(),0);
        for(int k=0;k<nums.size();k++){
            nums[k] = nums[k]*nums[k];
        }
        
        while(j>=i){
            if(nums[i]>=nums[j]){
                pums[index] = nums[i];
                i++;
            }
            else{
                pums[index] = nums[j];
                j--;
            }
            index--;
        }
        return pums;
    }
};
